### PR TITLE
fix: view logs not showing full screen height on load

### LINKED ISSUE #501: Nit: dead space under Terminal/Log should be same as background colour
### Bug description

When there are no logs the actual log output window only goes half way down. The Terminal window looks like it resizes in increments of the text height. In either case there is a section at the bottom that's a different colour than the terminal window itself. Please set this section to the same colour as the terminal window background so that it doesn't stand out.

Logs
Terminal

### Operating system

macOS 12.6

### Version

0.8.0

### Steps to reproduce

_No response_

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by cdrage**
fix: view logs not showing full screen height on load

### What does this PR do?

Fixes the issue of the logs only showing half the screen rather than
full.

This is caused by the xterm package incorrectly loading and setting the
height to half.

Related bug: https://github.com/xtermjs/xterm.js/issues/3887

### Screenshot/screencast of this PR

Before:
Screenshot 2022-11-24 at 10 13 44 AM

After:
Screenshot 2022-11-24 at 10 13 55 AM

### What issues does this PR fix or reference?

Closes https://github.com/containers/podman-desktop/issues/501

### How to test this PR?

Use `yarn watch` and load up the logs on a container. It should now be
full height.

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -94,6 +94,7 @@
     "moment": "^2.29.4",
     "os-locale": "^6.0.2",
     "stream-json": "^1.7.5",
+    "svelte-resize-observer": "^2.0.0",
```
Comment (benoitf) [inline package.json]: I'm not sure why we would need that dependency ?

ResizeObserver is in the API

Comment (cdrage) [reply package.json]: Good observation! I didn't see that it's actually in the API.  I'll update the PR.

Comment (benoitf) [reply package.json]: I think it's because there is a polyfill but we don't need it

File: packages/renderer/src/lib/ContainerDetailsLogs.svelte [hunk 2]
```diff
@@ -90,14 +93,35 @@ async function refreshTerminal() {
...
+  if (logsTerminal) {
+    termFit.fit();
+  }
+}
+
 onMount(async () => {
   refreshTerminal();
 });
+
+onMount(() => {
```
Comment (cdrage) [inline packages/renderer/src/lib/ContainerDetailsLogs.svelte]: Is is okay having two `onMount` in svelte? Unsure the most ideal way to implement this. @benoitf

**Comment by cdrage** [comment]
Updated! Thanks for the insight @benoitf 

Just need help with regards to if this is the best way for implementation.

File: packages/renderer/src/lib/ContainerDetailsLogs.svelte (hunk 2 above)
Comment (benoitf) [reply packages/renderer/src/lib/ContainerDetailsLogs.svelte]: better to use only one 👍  (especially the first one being really short)

File: packages/renderer/src/lib/ContainerDetailsLogs.svelte [hunk 3]
```diff
@@ -90,14 +93,33 @@ async function refreshTerminal() {
...
+  // Resize the terminal each time we change the div size
+  const resizeObserver = new ResizeObserver(entries => {
+    resizeTerminal();
+  });
+
+  resizeObserver.observe(logsXtermDiv);
+
+  // Cleanup the observer
+  return () => resizeObserver.unobserve(logsXtermDiv);
```
Comment (benoitf) [inline packages/renderer/src/lib/ContainerDetailsLogs.svelte]: You can also move the code to onDestroy() method to make it more visible (your code is still valid as onMount can return a function but it's less explicit)

**Comment by cdrage** [comment]
@benoitf Updated. Let me know if this is proper svelte standards / practice? Code seems to work well / I don't notice any issues.

File: packages/renderer/src/lib/ContainerDetailsLogs.svelte [hunk 4]
```diff
@@ -90,13 +94,34 @@ async function refreshTerminal() {
...
+      termFit.fit();
     }
   });
-  fitAddon.fit();
+  termFit.fit();
+}
+
+function resizeTerminal() {
+  if (logsTerminal) {
+    termFit.fit();
```
Comment (benoitf) [inline packages/renderer/src/lib/ContainerDetailsLogs.svelte]: it looks odd to check for logsTerminal and call termFit ?

File: packages/renderer/src/lib/ContainerDetailsLogs.svelte [hunk 5]
```diff
@@ -90,13 +94,34 @@ async function refreshTerminal() {
...
+    resizeTerminal();
+  });
+
+  // Observe the terminal div
+  resizeObserver.observe(logsXtermDiv);
+});
+
+onDestroy(() => {
+  // Cleanup the observer on destroy
+  resizeObserver.unobserve(logsXtermDiv);
```
Comment (benoitf) [inline packages/renderer/src/lib/ContainerDetailsLogs.svelte]:
```suggestion
  resizeObserver?.unobserve(logsXtermDiv);
```

File: packages/renderer/src/lib/ContainerDetailsLogs.svelte (hunk 4 above)
Comment (benoitf) [reply packages/renderer/src/lib/ContainerDetailsLogs.svelte]: you may remove the if
termFit?.fit();

File: packages/renderer/src/lib/ContainerDetailsLogs.svelte [hunk 6]
```diff
@@ -90,13 +94,34 @@ async function refreshTerminal() {
...
+  }
 }
+
 onMount(async () => {
+  // Refresh the terminal on initial load
   refreshTerminal();
+
+  // Resize the terminal each time we change the div size
+  resizeObserver = new ResizeObserver(entries => {
+    resizeTerminal();
```
Comment (benoitf) [inline packages/renderer/src/lib/ContainerDetailsLogs.svelte]: it seems the function is only called there, so probably we don't need the function and just call termFit?.fit();

**Comment by cdrage** [comment]
@benoitf thanks for the review! ready for another one

**Review by benoitf — APPROVED** [review]
I was never able to reproduce the issue
but on my side it works as before so 👍 

ready to be merged

**Pull request merged**